solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -315,8 +320,17 @@ public void runImpl(CommandLine cli) throws Exception {
         throw new IllegalArgumentException(
             "Must specify -c / --name parameter with --solr-url to post documents.");
       }
-      String url =
-          SolrCLI.normalizeSolrUrl(cli) + "/solr/" + cli.getOptionValue("name") + "/update";
+
+      String solrUrl = cli.getOptionValue("solr-url");
+
+      String hostContext = System.getProperty("hostContext", "/solr");
+      if (hostContext.isBlank()) {
+        log.warn("Invalid hostContext {} provided, setting to /solr", hostContext);
+        hostContext = "/solr";
+      }
+
+      solrUrl = SolrCLI.normalizeSolrUrl(solrUrl, true, hostContext) + hostContext;
```

Comment:
I thought of the same workaround, and guess I'm medium happy about it. Would of course prefer a `SolrCLI` util method that hides this such as `SolrCLI.getAndNormalizeSolrUrl(cli), and consider using it wherever we do raw `cli.getOptionValue("solr-url")`, since we'd potentially run into same issue elsewhere.